pkg/rules/rules_test.go

```diff
@@ -954,3 +955,361 @@ func TestDedupGroups(t *testing.T) {
 		})
 	}
 }
+
+func TestFilterRules(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		matcherSets  [][]*labels.Matcher
+		groups, want []*rulespb.RuleGroup
+	}{
```

Comment:
Just to be more resilient, it'd be great if you can add a couple more cases.
- Multiple rules with only templated labels, and non-templated matcherSets
- A basic case but with other type of matcher than just `MatchEqual`

Reply:
Yes, have added a test case with templated labels and non templated matcherSet as well as a regexp matcher test case!